src/main/java/com/github/splendor_mobile_game/game/model/Game.java

```diff
@@ -128,15 +138,16 @@ private ArrayList<Card> getRandomCards(ArrayList<Card> cardList, int amount) {
             list.add(i);
         }
 
+
         Random rand = new Random();
         while(list.size() > 0) {
             int index = rand.nextInt(list.size()); // Get random index
-            array.add(cardList.get(index));
+            array.add(deck.get(index));
+            //If the card was drawn it would not be used at any time by the Game class
+            deck.remove(index);
             System.out.println("Selected: " + list.remove(index));
```

Comment:
list.remove(index) shouldn't be inside the System.out.println() instruction. 
Did you test this function in case of appearing duplicates?

Reply:
This System print is not written by the @v4m3rrr. It was written before by someone.